pkgs/development/python-modules/opentelemetry-instrumentation-sqlalchemy/default.nix

```
@@ -0,0 +1,54 @@
+{
+  buildPythonPackage,
+  pythonOlder,
+
+  # build-system
+  hatchling,
+
+  # dependencies
+  aiosqlite,
+  opentelemetry-api,
+  opentelemetry-instrumentation,
+  opentelemetry-semantic-conventions,
+  packaging,
+  sqlalchemy,
+  wrapt,
+
+  # tests
+  opentelemetry-test-utils,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  inherit (opentelemetry-instrumentation) version src;
+  pname = "opentelemetry-instrumentation-sqlalchemy";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  sourceRoot = "${opentelemetry-instrumentation.src.name}/instrumentation/${pname}";
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    aiosqlite
+    opentelemetry-api
+    opentelemetry-instrumentation
+    opentelemetry-semantic-conventions
+    packaging
+    sqlalchemy
+    wrapt
+  ];
```

```suggestion
  dependencies = [
    opentelemetry-api
    opentelemetry-instrumentation
    opentelemetry-semantic-conventions
    packaging
    wrapt
  ];
  
  optional-dependencies = {
    instruments = [ sqlalchemy ];
  };
```
https://github.com/open-telemetry/opentelemetry-python-contrib/blob/2480f4e846286df1d0ae0a6a0bc7293bc9c88813/instrumentation/opentelemetry-instrumentation-sqlalchemy/pyproject.toml#L27-L33

---

I had `sqlalchemy` as a hard dependency because [it is required to import the module](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/2480f4e846286df1d0ae0a6a0bc7293bc9c88813/instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/__init__.py#L101). The import check won't succed otherwise. To be honest, it would be quite odd to have `sqlalchemy` as an optional dependency when it is part of the name of the package.